### PR TITLE
BVH acceleration and other modifications

### PR DESCRIPTION
**Pull request by Enigmatisms**
Marching towards v1.2.1, where we have our first BVH acceleration structure which is capable of accelerating ray intersection by reducing the amount of primitives to be taken into account. Test showed that, for `bunny` scene which contains 14916 triangles, it only takes 30 ms to build BVH tree while achieving roughly 50x speed up compared to the brute force ray intersection which can only prune unnecessary computation via object bounding box. The implementation of BVH tree is done in C++ and I expose a python API via [pybind/pybind11](https://github.com/pybind/pybind11). All it takes is to build the bvh code and install locally with minimal effort. 

Other modifications include: 
- Repo structure: clearer, renamed a few folders and did some moving.
- Temporary deprecated `transient_lit` mode, the problem of which can be found in issue #9 
- Re-organized primitive storage structure, a more compact and BVH-compatible dense structure is used.
- Fixed some other minor bugs.

### CODE REVIEW

File: assets/ti_tests/disk_sampling.py
Comment (hqyswife) [inline assets/ti_tests/disk_sampling.py:1]: Note that a git ignore file should be added in `asset/ti_tests/`

File: tracer/bvh_visualize.py
Comment (hqyswife) [inline tracer/bvh_visualize.py:1]: This should be implemented in the future

**Review by hqyswife — APPROVED** [review]
Well done!

**Comment by Enigmatisms** [comment]
Merging pull request and releasing v1.2.1

**Pull request merged**